src/main/java/com/example/weatherwaveapi/config/GeneralSettings.java

```diff
@@ -0,0 +1,44 @@
+package com.example.weatherwaveapi.config;
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@ConfigurationProperties(prefix = "application.settings")
+public class GeneralSettings {
+
+    private OpenWeather openWeatherMap;
+    private Yandex yandexMap;
+    private RestTemplate restTemplateProperties;
+
+    @Data
+    public static class OpenWeather {
+        private Api api;
+
+        @Data
+        public static class Api {
+            private String key;
+            private String weatherUrl;
+            private String forecastUrl;
+        }
+    }
+
+    @Data
+    public static class Yandex {
+        private Api api;
+
+        @Data
+        public static class Api {
```

Comment:
static classes with nested classes is a rather complex structure, it can bring us errors when used. Also, using the Date annotation for configuration classes is redundant. I would suggest changing the class to record and adjusting the settings according to the prefix from the Yaml file. Then we won't need any external dependencies.